pkg/doc/verifiable/json.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package verifiable
+
+import (
+	"encoding/json"
+)
+
+func marshalJSON(v interface{}, ef map[string]interface{}) ([]byte, error) {
+	// Convert value into a JSON map of known fields.
+	kf, err := valueToMap(v, ef)
+	if err != nil {
+		return nil, err
+	}
+
+	// Marshal extended known fields map.
+	return json.Marshal(kf)
+}
+
+func unmarshalJSON(data []byte, v interface{}) (map[string]interface{}, error) {
+	err := json.Unmarshal(data, v)
+	if err != nil {
+		return nil, err
+	}
+
+	// Collect known fields map.
+	vData, err := json.Marshal(v)
+	if err != nil {
+		return nil, err
+	}
+	var kf map[string]interface{}
+	err = json.Unmarshal(vData, &kf)
+	if err != nil {
+		return nil, err
+	}
+
+	// Collect all fields map.
+	ef := make(map[string]interface{})
+	err = json.Unmarshal(data, &ef)
+	if err != nil {
+		return nil, err
+	}
+
+	// Complement all fields with known fields to become a set of extra fields.
+	for k := range kf {
+		delete(ef, k)
+	}
+
+	return ef, nil
+}
+
+func valueToMap(v interface{}, ef map[string]interface{}) (map[string]interface{}, error) {
+	// Convert raw credential into a JSON map of known fields.
+	rcBytes, err := json.Marshal(v)
+	if err != nil {
+		return nil, err
+	}
+
+	var kf map[string]interface{}
+	err = json.Unmarshal(rcBytes, &kf)
+	if err != nil {
+		return nil, err
+	}
+
+	// Supplement value map with extra fields.
+	for k, v := range ef {
+		if _, exists := kf[k]; !exists {
+			kf[k] = v
+		}
+	}
+
+	return kf, nil
```

Comment:
same thing for the function name here.. it appends `ef` to `v`.. need a better name.. `valeToMap` doesn't convey this

Reply:
The name is changed to `mergeValueWithMap()`. What do you fill about it @Baha-sk ?

Reply:
as discussed @kdimak, you will change this name again to make it clearer ..

Reply:
done!